src/components/forms/TextInput/TextInput.tsx

```diff
@@ -50,21 +55,83 @@ export const TextInput = ({
       'usa-input--success': isSuccess,
       'usa-input--small': isSmall,
       'usa-input--medium': isMedium,
+      'usa-masked': mask,
     },
     className
   )
 
-  return (
+  const [inputValue, setInputValue] = useState('')
+  const input = (
     <input
       data-testid="textInput"
       className={classes}
       id={id}
       name={name}
       type={type}
       ref={inputRef}
+      maxLength={
+        inputProps.maxLength || !mask ? inputProps.maxLength : mask.length
+      }
+      onChange={mask ? handleChange : inputProps.onChange}
+      value={mask ? inputValue : inputProps.value}
       {...inputProps}
     />
   )
+
+  const [maskValue, setMaskValue] = useState(mask)
+  const [iValue, setIValue] = useState(<i></i>)
```

Comment:
**suggestion**: Perhaps the initial state of iValue should be `undefined` so we don't render empty tags. Thoughts?

Reply:
yeah, just requires setting the type more explicitly

Reply:
(update to this thread [here](https://github.com/trussworks/react-uswds/pull/2538#discussion_r1286289595))